src/core/backend/pearl/group.rs

```diff
@@ -342,3 +358,40 @@ impl Display for Group {
             .finish()
     }
 }
+
+struct PartitionName {
+    timestamp: u64,
+    hash: String,
+}
+
+impl PartitionName {
+    fn new(timestamp: u64, hash: &str) -> Self {
+        Self {
+            timestamp,
+            hash: hash.to_string(),
+        }
+    }
+
+    fn try_from_string(s: &str) -> Option<Self> {
+        let (timestamp_string, hash_string) = if let Some(index) = s.find('_') {
+            (&s[0..index], &s[index + 1..])
+        } else {
+            (s, "")
+        };
+        if let Ok(timestamp) = timestamp_string.parse() {
+            Some(Self {
+                timestamp,
+                hash: hash_string.to_string(),
+            })
+        } else {
+            warn!("Failed to parse timestamp from {}", timestamp_string);
```

Comment:
I think log message needed on higher level. Plus, you can write shorter: `timestamp_string.parse().ok()`.
I don't insist, on your consideration (: